lib/mutant/integration.rb

```diff
@@ -4,38 +4,23 @@ module Mutant
   class Integration
     include AbstractType, Adamantium::Flat, Concord.new(:config)
 
-    REGISTRY = {}
-
     # Setup integration
     #
-    # @param [String] name
+    # Integrations are supposed to define a constant under
+    # Mutant::Integration named after the capitalized +name+
+    # parameter.
     #
-    # @return [Integration]
-    def self.setup(name)
-      require "mutant/integration/#{name}"
-      lookup(name)
-    end
-
-    # Lookup integration for name
+    # This avoids having to maintain a mutable registry.
     #
-    # @param [String] name
+    # @param kernel [Kernel]
+    # @param name [String]
     #
-    # @return [Integration]
-    #   if found
-    def self.lookup(name)
-      REGISTRY.fetch(name)
+    # @return [Class<Integration>]
+    def self.setup(kernel, name)
+      kernel.require("mutant/integration/#{name}")
```

Comment:
For dependency injection reasons. I do not want to refer to any "IO global" in the mutant code base for various reasons:

* Its harder to test
* I can inject metric tracer objects on debugging
* I do want to inject objects that do something fundamentally different in future, these are in private branches right now, but that behavior depends on "perfect" hickacks of interaction with `Kernel`.